org.osgi.service.feature/src/org/osgi/service/feature/FeatureService.java

```diff
@@ -35,6 +35,49 @@
      */
     public BuilderFactory getBuilderFactory();
 
+	/**
+	 * Obtain an ID from a Maven ID string.
+	 * 
+	 * @param mavenID The Maven ID.
+	 * @return the ID.
+	 */
+	public ID getIDfromMavenID(String mavenID);
+
+	/**
+	 * Obtain an ID.
+	 * 
+	 * @param groupId The group ID.
+	 * @param artifactId The artifact ID.
+	 * @param version The version.
+	 * @return The ID.
+	 */
+	public ID getID(String groupId, String artifactId, String version);
+
+	/**
+	 * Obtain an ID.
+	 * 
+	 * @param groupId The group ID.
+	 * @param artifactId The artifact ID.
+	 * @param version The version.
+	 * @param type The type.
+	 * @return The ID.
+	 */
+	public ID getID(String groupId, String artifactId, String version,
+			String type);
+
+	/**
+	 * Obtain an ID.
+	 * 
+	 * @param groupId The group ID.
+	 * @param artifactId The artifact ID.
+	 * @param version The version.
+	 * @param type The type.
+	 * @param classifier The classifier.
+	 * @return The ID.
+	 */
+	public ID getID(String groupId, String artifactId, String version,
```

Comment:
Type must not be null is classifier is not null.

Reply:
Hi @stbischof there are 3 overloads for `getID()` - none of them should be used with `null` arguments but they express what is allowable through the method signature.

I also dded a constant for the Feature Type into the ID interface. I'm not sure an extra method to create IDs for features adds much value if we already have the constant.

Reply:
that is enough.

Reply:
last edge-case:
- What happens with empty values or if they contain a ":"?

Reply:
For each argument, the javadoc should state whether the value must not be null or may be null. Then it is clear to the reader and the TCK implementor.

Reply:
> last edge-case:
> 
> * What happens with empty values or if they contain a ":"?

Would that not be outside the grammar definition?

    groupId ':' artifactId ( ':' type ( ':' classifier )? )? ':' version

which says: a ':' is _always_ followed by a value, or am I misunderstanding?

Reply:
I guess it should be clear that the productions for `groupId` etc. result in non-empty tokens.